src/devices/cpu/ht1130/ht1130.h

```diff
@@ -0,0 +1,131 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+
+#ifndef MAME_CPU_HT1130_HT1130_H
+#define MAME_CPU_HT1130_HT1130_H
+
+#pragma once
+
+
+class ht1130_device : public cpu_device
+{
+public:
+	ht1130_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+
+	enum {
+		HT1130_PC,
+		HT1130_R0,
+		HT1130_R1,
+		HT1130_R2,
+		HT1130_R3,
+		HT1130_R4,
+		HT1130_ACC,
+		HT1130_TIMER_EN,
+		HT1130_TIMER,
+	};
+
+	auto pm_in_cb() { return m_port_in_pm.bind(); }
+	auto ps_in_cb() { return m_port_in_ps.bind(); }
+	auto pp_in_cb() { return m_port_in_pp.bind(); }
+
+	auto pa_out_cb() { return m_port_out_pa.bind(); }
+
+	auto display_data_out_cb() { return m_display_data_out.bind(); }
+
+	void wake() // can be configured to wake from certain inputs as a mask option?
+	{
+		machine().scheduler().synchronize();
+		m_inhalt = 0;
+	}
```

Comment:
This won’t work – `synchronize` takes effect asynchronously, so the change will still be visible before syncrhonisation takes place if you do it like this.  You need to provide a callback to `synchronize` to do the actual work.

Reply:
I think I'll just back out the 'wake' changes instead

Reply:
You’ve been around long enough to know how `synchronize` works, and there are numerous examples of it in the code (e.g. Namco 06xx/50xx/51xx/52xx/54xx, the devices used for sound latches, the input merger devices, and Apple II Centronics printer cards, just off the top of my head).

Reply:
I've backed out the changes, I'm not happy with that part anyway.

Reply:
For CPU devices synchronized inputs, you can use set_input_line instead. Just like you wouldn't do irq_w(x) but instead set_input_line(x ? ASSERT_LINE : CLEAR_LINE)

Reply:
well for now it isn't really important.  it can be added later if needs be.

Reply:
Hap's correct, if you just use an input line it's synchronized for you.

Reply:
I've already said, I've backed out those changes, so it's irrelevant at this point.